text/0141-linking-traces.md

```diff
@@ -0,0 +1,368 @@
+# Motivation
+
+At the time of writing this RFC, Sentry (the product as well as SDKs) cannot connect multiple traces to deliver a "bigger picture" of what happened before (or after) a specific trace. 
+A trace is therefore the highest level of connecting signals (errors, traces, profiles, etc.).
+The notable exception is Session Replay, where we are indeed able to provide this bigger picture.
+
+This RFC proposes a method of achieving linkage between (and within) traces in the entire Sentry product, to satisfy a number of currently sub-optimally solved use cases.
+
+## Frontend Traces and User Journeys
+
+The most important application of linking traces are frontend applications. We would like to display a user journey (session) to make debugging of issues easier as developers get more context on what happened before a specific issue. 
+Today, we are limited to the duration of one trace (id), which is handled differently and kept alive for different times across our various SDKs. 
+
+A concrete example are our [JavaScript Browser SDKs](https://develop.sentry.dev/sdk/platform-specifics/javascript-sdks/browser-tracing/#tracing-model) which by default keep a trace alive as long as users
+are on the same page or (URL) route. This was a compromise in which we accepted that a trace would consist of multiple trace root spans (transactions), which is generally discouraged by tracing models like OpenTelemetry.
+
+Another example of a suboptimal trace model is the one used in most mobile SDKs. In these SDKs, traces are mostly started via idle transactions, meaning transactions start a fixed point but end automatically after a specific period of inactivity (i.e. no child spans being added). Errors occurring while no transaction is active are associated with a "fallback" `traceId` stored on the global SDK scope that stays the same until the SDK is again initialized. The consequence is that potentially hundreds of unrelated events are associated with the same fallback trace as [outlined](https://github.com/getsentry/rfcs/blob/rfc/mobile-tracing-without-performance-v-2/text/0136-mobile-tracing-without-performance-v-2.md) in a previous attempt to improve this behavior.
+
+## Queues and Async Operations
+
+Somewhat related, we also face situations in which child spans are not started and finished within the time span of their parent span. As per OpenTelemetry specification, this violates the integrity of a trace. A concrete example for this is that we [currently recommend](https://docs.sentry.io/platforms/javascript/guides/node/tracing/instrumentation/custom-instrumentation/queues-module/) users to create one trace for a producer-consumer (Queue) cycle, where the spans from the consumer likely start after the producer spans finished. OpenTelemetry in fact [recommends](https://opentelemetry.io/docs/concepts/signals/traces/#span-links) to start separate traces for the consumer, producer or more generally async operations and to link these traces via a span link.
```

Comment:
> Somewhat related, we also face situations in which child spans are not started and finished within the time span of their parent span. As per OpenTelemetry specification, this violates the integrity of a trace.

It doesn't really affect the proposal but I think that might be stated a bit too strongly - I've never found a place where it's forbidden by the spec for a child to outlive its parent, and the spec even [has an example ASCII doc showing such a case](https://opentelemetry.io/docs/specs/otel/overview/#traces). But, as you say, the convention does seem to be to use span links instead of parent/child relationships for async operations.

(My personal opinion is also that it's better modelling practice to use links for async children and I'd rather we go in that direction, but that's just me 😁)

Reply:
I reworded this section. Thanks for clarifying! I must have misheard this somewhere then.